Sources/Widgets/Representations/LineHandleRepresentation/index.js

```diff
@@ -92,91 +62,50 @@ function vtkLineHandleRepresentation(publicAPI, model) {
     model.displayMapper.setCallback(callback ? callbackProxy : null);
   };
 
-  const superGetRepresentationStates = publicAPI.getRepresentationStates;
-  publicAPI.getRepresentationStates = (input = model.inputData[0]) =>
-    superGetRepresentationStates(input).filter(
-      (state) => state.getOrigin?.() && state.isVisible?.()
-    );
-
-  // --------------------------------------------------------------------------
-
-  publicAPI.requestData = (inData, outData) => {
-    const list = publicAPI.getRepresentationStates(inData[0]);
-    const totalCount = list.length;
-
-    const points = publicAPI
-      .allocateArray('points', 'Float32Array', 3, totalCount)
-      .getData();
-    const color = publicAPI
-      .allocateArray('color', ...publicAPI.computeColorType(list), totalCount)
-      .getData();
-    const scale = publicAPI
-      .allocateArray('scale', 'Float32Array', 3, totalCount)
-      .getData();
-    const direction = publicAPI
-      .allocateArray('direction', 'Float32Array', 9, totalCount)
-      .getData();
-
-    for (let i = 0; i < totalCount; i++) {
-      const state = list[i];
-      const isActive = state.getActive();
-      let scaleFactor = isActive ? model.activeScaleFactor : 1;
-
-      const coord = state.getOrigin();
-      points[i * 3 + 0] = coord[0];
-      points[i * 3 + 1] = coord[1];
-      points[i * 3 + 2] = coord[2];
-
-      const right = state.getRight ? state.getRight() : [1, 0, 0];
-      const up = state.getUp ? state.getUp() : [0, 1, 0];
-      const dir = state.getDirection ? state.getDirection() : [0, 0, 1];
-      const rotation = [...right, ...up, ...dir];
-      direction.set(rotation, 9 * i);
-
-      let scale3 = state.getScale3 ? state.getScale3() : [1, 1, 1];
-      scale3 = scale3.map((x) => (x === 0 ? model.defaultScale : x));
-      if (model.infiniteLine) {
-        scale3[2] = 100000;
-      }
-
-      const scale1 = state.getScale1 ? state.getScale1() : model.defaultScale;
-      if (publicAPI.getScaleInPixels()) {
-        scaleFactor *= getPixelWorldHeightAtCoord(
-          coord,
-          model.displayScaleParams
-        );
+  /**
+   * Overwrite scale3 to optionally make lines infinite
+   */
+  const superScale3 = publicAPI.getScale3();
+  publicAPI.setScale3((polyData, states) => {
+    superScale3(polyData, states);
+    if (model.infiniteLine) {
+      const scales = allocateArray(
+        polyData,
+        'scale',
+        states.length,
+        'Float32Array',
+        3
+      ).getData();
+      for (let i = 0; i < states.length; ++i) {
+        scales[3 * i + 2] = 100000;
```

Comment:
Let's turn this magic "infinity" number into a named constant for clarity. Same for the other instance of it that references "infinite line".